db/models.py

```diff
@@ -50,3 +52,61 @@ class MovieSession(models.Model):
 
     def __str__(self) -> str:
         return f"{self.movie.title} {str(self.show_time)}"
+
+
+class Order(models.Model):
+    created_at = models.DateTimeField(auto_now_add=True)
+    user = models.ForeignKey("User", on_delete=models.CASCADE)
+
+    def __str__(self) -> str:
+        return str(self.created_at)
+
+    class Meta:
+        ordering = ("-created_at",)
+
+
+class Ticket(models.Model):
+    movie_session = models.ForeignKey("MovieSession", on_delete=models.CASCADE)
+    order = models.ForeignKey("Order", on_delete=models.CASCADE)
+    row = models.IntegerField()
+    seat = models.IntegerField()
+
+    class Meta:
+        constraints = [models.UniqueConstraint(
+            fields=["movie_session", "row", "seat"],
+            name="unique_ticket"
+        )]
+
+    def save(self, *args, **kwargs) -> None:
+        self.full_clean()
+        super().save(*args, **kwargs)
+
+    def clean(self) -> None:
+        if self.movie_session:
+            if not 1 <= self.row <= self.movie_session.cinema_hall.rows:
+                raise ValidationError(
+                    {
+                        "row": [f"row number must be in available range: "
+                                f"(1, rows): "
+                                f"(1, {self.movie_session.cinema_hall.rows})"]
+                    }
+                )
+
+            if not 1 < self.seat < self.movie_session.cinema_hall.seats_in_row:
```

Comment:
There is an off-by-one error in the seat validation logic. The condition should be `1 <= self.seat <= self.movie_session.cinema_hall.seats_in_row` to include both the first and last seat in the range.